doc/source/file_formats.rst

```diff
@@ -1,23 +1,545 @@
+      .. code-block:: 
+
+        [ link ]
+        [ bonds ]
+        BB +BB  1 0.47 5000
+
+- ``[ patterns ]``
+    - ``optional``
+    - If no pattern is given, the link pattern will consists of the
+      particles and their connectivity inferred from the interactions
+      and atoms sub directive.
+    - To overwrite this default pattern one can list patterns of atoms
+      to which the link applies. Each line in the subsection describes
+      a pattern. One of the patterns must apply for the link to match.
+      A pattern consists of atom identifiers. Each atom identifier
+      consists of a name which may be preceded by a prefix indicating
+      the relative position in terms of residues.
+
+- ``[ features]``
+    - ``optional``
+    - The features subsection lists features to apply to the link
+      itself. These can be used to control the application of links
+      during the building of topologies. For example, setting the
+      feature 'scfix' will cause the links to be applied only if the
+      option -scfix is given to martinize2.
+
+- ``[ molmeta ]``
+    - ``optional``
+    - The molmeta subsection lists metadata to be added/changed in the
+      molecular graph. These metadata can be used (and modified) by
+      Vermouth's processors and for provenance.
+
+- ``[ edges ]``
+    - ``optional``
+    - Within the context of links, the edges specify that an
+      edge should or shouldn't be present, respectively, for the link
+      pattern to match.
+    - They should mostly be used in cases, where interactions are
+      applied for which edges cannot be made automatically.
+
+- ``[ non-edges ]``
+    - ``optional``
+    - This directive specifies that an edge should be absent in order
+      for the link to apply. Note that the first particle must be
+      present in the link and the second one must the partner
+      with which to not form an edge.
+    - This syntax is likely to be deprecated in the near future.
+
+
+Allowed sub-directives: Modifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- ``[ atoms ]``
+    - ``mandatory``
+    - The atoms subsection under a modification lists both anchors and
+      atoms to be added to anchors or changed. Entries consist of an
+      atom name followed by an attributes statement. Atoms that are
+      added need to set the "PTM_atom" attribute to true and require
+      a valid "element" attribute. The "replace" attribute may be set
+      to a (nested) JSON dict, listing the atom attributes to be
+      changed and the new values corresponding to the modification.
+      Such changes can also be applied to atoms already present in the
+      molecular graph, i.e., the 'non-PTM atoms'.`
+- ``[ interaction_name ]``
+    - ``optional``
+    - A link may list any number of interactions to be added, if a
+      modification applies. The syntax is the same as for the
+      link sub directive.
+
+Special meta data
+^^^^^^^^^^^^^^^^^
```

Comment:
```suggestion
Special meta data
~~~~~~~~~~~~~~~~
```
Otherwise it becomes a subsection of the modification subsections. Also, check the number of ~, I eyeballed it.

Reply:
done; but was missing one ~